python/ray/serve/_private/deployment_info.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any, Dict, Optional
+
+import ray
+from ray.serve._private.autoscaling_policy import BasicAutoscalingPolicy
+from ray.serve._private.common import TargetCapacityDirection
+from ray.serve._private.config import DeploymentConfig, ReplicaConfig
+from ray.serve.generated.serve_pb2 import DeploymentInfo as DeploymentInfoProto
+from ray.serve.generated.serve_pb2 import (
+    TargetCapacityDirection as TargetCapacityDirectionProtoEnum,
+)
+
+# Concurrency group used for operations that cannot be blocked by user code
+# (e.g., health checks and fetching queue length).
+CONTROL_PLANE_CONCURRENCY_GROUP = "control_plane"
+REPLICA_DEFAULT_ACTOR_OPTIONS = {
+    "concurrency_groups": {CONTROL_PLANE_CONCURRENCY_GROUP: 1}
+}
+
+
+TARGET_CAPACITY_DIRECTION_TO_PROTO = {
+    None: TargetCapacityDirectionProtoEnum.UNSET,
+    TargetCapacityDirection.UP: TargetCapacityDirectionProtoEnum.UP,
+    TargetCapacityDirection.DOWN: TargetCapacityDirectionProtoEnum.DOWN,
+}
+
+
+TARGET_CAPACITY_DIRECTION_FROM_PROTO = {
+    TargetCapacityDirectionProtoEnum.UNSET: None,
+    TargetCapacityDirectionProtoEnum.UP: TargetCapacityDirection.UP,
+    TargetCapacityDirectionProtoEnum.DOWN: TargetCapacityDirection.DOWN,
+}
```

Comment:
Why do we need to have both enums? Can we instead just use the proto one?

Reply:
I simplified the conversion. I pattern-matched `DeploymentStatus`, which also uses a separate enum for the protobuf and the Python code.